Description: Enables gstreamer camerabin backend.
Adds mediacontroller videosource to Gstreamer camerabin backend.
Author: Joonas Tanskanen <joonas.tanskanen@sasken.com>
--- a/plugins/multimedia/gstreamer/camerabin/camerabinaudioencoder.cpp
+++ b/plugins/multimedia/gstreamer/camerabin/camerabinaudioencoder.cpp
@@ -49,7 +49,7 @@ CameraBinAudioEncoder::CameraBinAudioEnc
 {
     QList<QByteArray> codecCandidates;
 
-#if defined(Q_WS_MAEMO_5) || defined(Q_WS_MAEMO_6)
+#if defined(Q_WS_MAEMO_5) || defined(Q_WS_MAEMO_6) || defined(Q_WS_MEEGO)
     codecCandidates << "audio/AAC" << "audio/PCM" << "audio/AMR" << "audio/AMR-WB" << "audio/speex"
                     << "audio/ADPCM" << "audio/iLBC" << "audio/vorbis" << "audio/mpeg" << "audio/FLAC";
 
--- a/plugins/multimedia/gstreamer/camerabin/camerabinvideoencoder.cpp
+++ b/plugins/multimedia/gstreamer/camerabin/camerabinvideoencoder.cpp
@@ -49,7 +49,7 @@ CameraBinVideoEncoder::CameraBinVideoEnc
     :QVideoEncoderControl(session), m_session(session)
 {
     QList<QByteArray> codecCandidates;
-#if defined(Q_WS_MAEMO_5)
+#if defined(Q_WS_MAEMO_5) || defined(Q_WS_MEEGO)
     codecCandidates << "video/mpeg4" << "video/h264" << "video/h263" << "video/theora"
                     << "video/mpeg2" << "video/mpeg1" << "video/mjpeg" << "video/VP8" << "video/h261";
 
--- a/plugins/multimedia/gstreamer/gstreamer.pro
+++ b/plugins/multimedia/gstreamer/gstreamer.pro
@@ -52,6 +52,10 @@ maemo6 {
     }
 }
 
+meego {
+  PKGCONFIG +=gstreamer-plugins-bad-free-0.10
+}
+
 # Input
 HEADERS += \
     qgstreamermessage.h \
--- a/plugins/multimedia/gstreamer/camerabin/camerabinsession.cpp
+++ b/plugins/multimedia/gstreamer/camerabin/camerabinsession.cpp
@@ -383,6 +383,7 @@ GstElement *CameraBinSession::buildVideo
     } else {
         QList<QByteArray> candidates;
         candidates << "subdevsrc"
+                   << "mcsrc"
                    << "v4l2camsrc"
                    << "v4l2src"
                    << "autovideosrc";
